flake.nix

```
@@ -0,0 +1,131 @@
+        wallet = let pkgs = nixpkgsFor.${system};
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "bitcoin-transactions" ''
+                ${pkgs.xdg-utils}/bin/xdg-open ${pkgs.bitcoin-transactions}/html/wallet.html
+              ''
+            }/bin/bitcoin-transactions";
+        };
+        # This shows the Javascript unminified and readable
+        wallet-clear = let pkgs = nixpkgsFor.${system};
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "bitcoin-transactions" ''
+                ${pkgs.xdg-utils}/bin/xdg-open ${pkgs.bitcoin-transactions}/html/walletclear.html
+              ''
+            }/bin/bitcoin-transactions";
+        };
+      });
+
+      defaultPackage =
+        forAllSystems (system: self.packages.${system}.bitcoin-transactions);
+
+      devShell = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) mkShell nodePackages;
+        in mkShell {
+          buildInputs = [ nodePackages.browserify nodePackages.terser ];
+          inputsFrom = builtins.attrValues self.packages.${system};
+        });
+
+      # Tests run by 'nix flake check' and by Hydra.
+      checks = forAllSystems (system: {
+        inherit (self.packages.${system}) bitcoin-transactions;
+
+        # Additional tests, if applicable.
+        test = with nixpkgsFor.${system};
+          stdenv.mkDerivation {
+            name = "bitcoin-transactions-test-${version}";
+
+            buildInputs = [ nodejs bitcoin-transactions ];
+
+            unpackPhase = "true";
+
+            buildPhase = ''
+              echo 'running some integration tests'
+              ln -s ${bitcoin-transactions}/* .
+
+              # Run all general tests apart from the one that checks the network connection
+              while read line; do
+                if [[ $line == node* ]] && [[ $line != *suprnova* ]]; then
```

Nitpick: quotes around `$line` is good practice. Otherwise you'll hit an edge case (perhaps not in this scenario but in general) because  `line` is an empty string. Then you'll have something like `if [[ == node* ]] ...`. This will fail with a syntax error, instead of returning false, which is what you want.